src/main/java/edu/kit/provideq/toolbox/meta/Problem.java

```diff
@@ -82,6 +85,28 @@ public Mono<Solution<ResultT>> solve() {
         });
   }
 
+  public Mono<Bound> estimateBound() {
+    if (this.input == null) {
+      throw new IllegalStateException("Cannot estimate bound without input!");
+    }
+    
+    Function<InputT, Bound> estimator;
+    if (this.type.getEstimator().isEmpty()) {
```

Comment:
When you call `getEstimator()` you might get two different estimator optionals each time so it would be best if you save the result once and check `isEmpty()` and call `get()` on that optional. That will also make SonarLint happy :)